web/src/components/apps/Dashboard.jsx

```diff
@@ -26,65 +26,106 @@ const COMMON_ERRORS = {
+  const [activeChart, setActiveChart] = useState(null);
+  const [crosshairValues, setCrosshairValues] = useState([]);
+  const [noUpdatesAvalable, setNoUpdatesAvalable] = useState(false);
+  const [updateChecker, setUpdateChecker] = useState(updateCheckerRepeater);
+  const [uploadingAirgapFile, setUploadingAirgapFile] = useState(false);
+  const [airgapUploader, setAirgapUploader] = useState(null);
+  const [airgapUploadError, setAirgapUploadError] = useState(null);
+  const [viewAirgapUploadError, setViewAirgapUploadError] = useState(false);
+  const [viewAirgapUpdateError, setViewAirgapUpdateError] = useState(false);
+  const [airgapUpdateError, setAirgapUpdateError] = useState("");
+  const [startSnapshotErr, setStartSnapshotErr] = useState(false);
+  const [snapshotError, setSnapshotError] = useState(false);
+  const [startSnapshotErrorMsg, setStartSnapshotErrorMsg] = useState("");
+  const [showAutomaticUpdatesModalState, setShowAutomaticUpdatesModalState] =
+    useState(false);
+  const [showAppStatusModal, setShowAppStatusModal] = useState(false);
+  const [dashboard, setDashboard] = useState({
+    appStatus: null,
+    metrics: [],
+    prometheusAddress: "",
+  });
+  const [getAppDashboardJob, setGetAppDashboardJob] = useState(
+    getAppDashboardJobRepeater
+  );
+  const [fetchAppDownstreamJob, setFetchAppDownstreamJob] = useState(
+    fetchAppDownloadstreamJobRepeater
+  );
+  const [gettingAppLicenseErrMsg, setGettingAppLicenseErrMsg] = useState("");
+  const [startSnapshotOptions, setStartSnapshotOptions] = useState([
+    { option: "partial", name: "Start a Partial snapshot" },
+    { option: "full", name: "Start a Full snapshot" },
+    { option: "learn", name: "Learn about the difference" },
+  ]);
+  const [selectedSnapshotOption, setSelectedSnapshotOption] = useState({
+    option: "full",
+    name: "Start a Full snapshot",
+  });
+  const [snapshotDifferencesModal, setSnapshotDifferencesModal] =
+    useState(false);
+  const [gettingAppErrMsg, setGettingAppErrMsg] = useState("");
+  const [uploadProgress, setUploadProgress] = useState(0);
+  const [uploadSize, setUploadSize] = useState(0);
+  const [uploadResuming, setUploadResuming] = useState(false);
+  const [loadingApp, setLoadingApp] = useState(false);
+  const [displayErrorModal, setDisplayErrorModal] = useState(false);
+  const [startingSnapshot, setStartingSnapshot] = useState(false);
+  const [cluster, setCluster] = useState(clusterProp);
+
+  useEffect(() => {
+    if (app) {
+      setWatchState(app);
+      getAppLicense(app);
     }
-  }
+  }, [app]);
+
+  useEffect(() => {
+    setCluster(clusterProp);
+  }, [clusterProp]);
+
+  const setWatchState = (app) => {
+    setAppName(app.name);
+    setIconUri(app.iconUri);
+    setCurrentVersion(app?.downstream.currentVersion);
+    setDownstream(app.downstream);
+    setLinks(app.downstream?.links);
+  };
 
-  getAppLicense = async (app) => {
+  const getAppLicense = async (app) => {
```

Comment:
assigning all of this functions to const on every render will be inefficient but it should be fine for now.